Include/cpython/unicodeobject.h

```diff
@@ -183,6 +183,8 @@ PyAPI_FUNC(int) _PyUnicode_CheckConsistency(
 /* Interning state. */
 #define SSTATE_NOT_INTERNED 0
 #define SSTATE_INTERNED_MORTAL 1
+#define SSTATE_INTERNED_IMMORTAL 2
+#define SSTATE_INTERNED_IMMORTAL_STATIC 3
```

Comment:
Would you update this whatsnew topic?
https://github.com/python/cpython/blob/79311cbfe718f17c89bab67d7f89da3931bfa2ac/Doc/whatsnew/3.12.rst?plain=1#L951-L953

Reply:
@methane will do! Trying to get this to a steady state with all the CI passing, turns out that 6 months of rebasing broke a lot of things 😅 . After which, I will go through and update all the required documentation